Hunting Queries/MultipleDataSources/StorageAlertCorrelationwithCommonSecurityLogsandAuditLogs.yaml

```diff
@@ -0,0 +1,56 @@
+id: 860a8df2-8d19-4c60-bf61-de1c02422797
+name: Storage Alerts Correlation with CommonSecurityLogs & AuditLogs
+description: |
+  'This query combines different Storage alerts with CommonSecurityLogs and AuditLogs helping analysts /investigate any possible storage related attacks faster
+   thus reducing MTTR'
+requiredDataConnectors:
+  - connectorId: Fortinet
+    dataTypes:
+      - CommonSecurityLog
+  - connectorId: AzureSecurityCenter
+    dataTypes:
+      - SecurityAlert (ASC)
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - AuditLogs
+tactics:
+  - InitialAccess
+  - Impact
+relevantTechniques:
+  - T1190
+  - T1078
+query: |
+ let timeframe = 1d;
+  SecurityAlert
+  | where AlertName has_any ('Access from  a suspicious IP to a storage file share','	Access from a Tor exit node to a storage blob container')
+  | extend TimeGenerated1 = TimeGenerated
+  | extend EntitiesDynamicArray = parse_json(Entities) | mv-expand EntitiesDynamicArray
+  // Parsing relevant entity column extract hostname and IP address
+  | extend EntityType = tostring(parse_json(EntitiesDynamicArray).Type), EntityAddress = tostring(EntitiesDynamicArray.Address)
+  | extend IpAddress = iif(EntityType == 'ip', EntityAddress, '')
+   | where isnotempty(IpAddress) 
+  | join kind=inner (
+  CommonSecurityLog 
+  | where DeviceVendor has "Fortinet"
+  | where ApplicationProtocol has_any ("SSL","RDP")
+  | where LogSeverity has_any ("2","3")
+  | where isnotempty(SourceIP) and isnotempty(DestinationIP) and SourceIP != "0.0.0.0"
+  | where DeviceAction !in ("close", "client-rst", "server-rst", "deny") and DestinationPort != 161
+  | extend TimeGenerated2 = TimeGenerated
```

Comment:
Maybe a more descriptive name would be good here - such as NetworkEventTimeGenerated